Packet++/src/SingleCommandTextProtocol.cpp

```diff
@@ -5,29 +5,46 @@
 
 #define ASCII_HYPHEN 0x2d
 #define ASCII_SPACE 0x20
+#define MAX_COMMAND_LENGTH 9 // From SMTP command "STARTTLS" + 1 byte hyphen or space
+#define MIN_PACKET_LENGTH 2 // CRLF
 
 namespace pcpp
 {
 
 	size_t SingleCommandTextProtocol::getArgumentFieldOffset() const
 	{
 		size_t maxLen;
-		if (m_DataLen < 5)
+		if (m_DataLen < MAX_COMMAND_LENGTH)
 			maxLen = m_DataLen;
 		else
-			maxLen = 5;
-
-		// Find <SP> if exists
-		uint8_t *pos = (uint8_t *)memchr(m_Data, ASCII_SPACE, maxLen);
-		if (pos)
-			return pos - m_Data;
-
-		// Find Hyphen "-" if exists
-		pos = (uint8_t *)memchr(m_Data, ASCII_HYPHEN, maxLen);
-		if (pos)
-			return pos - m_Data;
-
-		return m_DataLen - 1;
+			maxLen = MAX_COMMAND_LENGTH;
+
+		// To correctly detect multi-line packets with the option containing a space in
+		// the first MAX_CONTENT_LENGTH bytes, search the both of hyphen and space to take
+		// correct command delimiter
+
+		std::string field(reinterpret_cast<char *>(m_Data), maxLen);
+
+		size_t posHyphen = field.find_first_of(ASCII_HYPHEN);
+		size_t posSpace = field.find_first_of(ASCII_SPACE);
+		size_t posCRLF = field.rfind("\r\n");
+
+		// No delimiter or packet end
+		if (posHyphen == std::string::npos && posSpace == std::string::npos && posCRLF == std::string::npos)
+			return 0;
+		// Hyphen not found while <SP> exists or both found but <SP> detected earlier
+		else if ((posHyphen == std::string::npos && posSpace != std::string::npos) ||
+				 (posSpace != std::string::npos && posHyphen != std::string::npos && posSpace < posHyphen))
+			return posSpace;
+		// <SP> not found while hyphen exists or both found but hyphen detected earlier
+		else if ((posSpace == std::string::npos && posHyphen != std::string::npos) ||
+				 (posSpace != std::string::npos && posHyphen != std::string::npos && posHyphen < posSpace))
+			return posHyphen;
```

Comment:
maybe a slightly simpler way to write it can be:
```cpp
else if (posHyphen != std::string::npos || posSpace != std::string::npos)
{
	return std::min(posSpace, posHyphen);
}
```

This should work since `npos` should be the max value representable by `size_t`